src/main/java/com/iotiq/application/service/SellerService.java

```diff
@@ -25,29 +27,38 @@ public class SellerService {
     private final OrderedProductService orderedProductService;
 
     @Transactional
-    public Seller getCurrentSellerOrCreate() {
+    public Seller getCurrentSeller() {
         User currentUser = userService.getCurrentUser();
-        return sellerRepository.findByUser(currentUser).orElseGet(() -> createDefaultSeller(currentUser));
+        return sellerRepository.findByUser(currentUser).orElseThrow(() -> new EntityNotFoundException(Seller.ENTITY_NAME));
     }
 
     @Transactional
     public SellerDto getSeller() {
-        SellerDto sellerDto = ModelMapperUtil.map(getCurrentSellerOrCreate(), SellerDto.class);
+        SellerDto sellerDto = ModelMapperUtil.map(getCurrentSeller(), SellerDto.class);
         sellerDto.setOrderedProducts(getLastTwoOrder());
         return sellerDto;
     }
 
     @Transactional
-    public Seller createDefaultSeller(User currentUser) {
+    public void createIfNotExists(User currentUser) {
+        Optional<Seller> seller = sellerRepository.findByUser(currentUser);
+
+        if(seller.isEmpty()) {
+            createSeller(currentUser);
+        }
```

Comment:
This might be better because you don't actually need the seller here.
```suggestion
        exists = sellerRepository.existsByUser(currentUser);

        if(!exists) {
            createSeller(currentUser);
        }
```